## 0x05. Python - Exceptions
This directory contains Python scripts that demonstrate the use of exceptions in Python. The project aims to familiarize students with exception handling, raising exceptions, and implementing safe operations in Python programming.

Key concepts covered in this project include:
- Try-except blocks for exception handling
- Raising exceptions
- Handling multiple exceptions
- Using finally clauses
- Implementing safe operations on lists and integers

The scripts in this directory showcase various operations such as safely printing list elements, handling division by zero, raising custom exceptions, and executing functions safely. They demonstrate how to use exception handling to make code more robust and handle potential errors gracefully.

This project builds upon basic Python knowledge and introduces more advanced concepts related to error handling and program flow control.

| File Name | Description     |
| ------------ | ------------    |
| `0-safe_print_list.py`              | Safely prints all elements of a list.                                      |
| `1-safe_print_integer.py`           | Safely prints an integer.                                                  |
| `2-safe_print_list_integers.py`     | Safely prints only the integers in a list.                                  |
| `3-safe_print_division.py`          | Safely divides two numbers and prints the result.                           |
| `4-list_division.py`                | Divides the corresponding elements of two lists and handles exceptions.     |
| `5-raise_exception.py`              | Raises a predefined exception.                                              |
| `6-raise_exception_msg.py`          | Raises an exception with a custom error message.                            |
| ----------------------------------- | ------------------------------------------------------------------------- |
| `100-safe_print_integer_err.py`     | Safely prints an integer and returns an error flag if unsuccessful.         |
| `101-safe_function.py`              | Executes a function and handles any exceptions that may occur.              |